paseos/communication/check_communication_link.py

```diff
@@ -0,0 +1,48 @@
+"""This file contains models of the communication transmission through a channel"""
+
+import pykep as pk
+from loguru import logger
+
+
+def check_communication_link(
+    local_actor,
+    local_actor_communication_link_name,
+    actor,
+    dt: float,
+    t0: float,
+    data_to_send_in_b: int,
+    window_timeout_value_in_s=7200
+):
+    """Checks how much data can be transmitted over the communication link and the length of the communication link.
+
+    Args:
+        local_actor (base_actor): local actor.
+        local_actor_communication_link_name (base_actor):  name of the local_actor's communication link to use.
+        actor (base_actor): other actor.
+        dt (float): simulation timestep [s].
+        t0 (float): current simulation time [s].
+        data_to_send_in_b (int): amount of data to transmit [b].
+        window_timeout_value_in_s (float, optional): timeout for estimating the communication window. Defaults to 7200.0.
+    Returns:
+        int: remaining amount of data to transmit [b].
+        float: length of the remaining communication window [s].
+    """
+    local_actor_comm_link = local_actor._communication_links[
+        local_actor_communication_link_name
+    ]
+
+    assert local_actor_comm_link.bandwidth_in_kbps > 0, "Bandiwidth has to be positive."
+
+    transmitted_data_b = 0
+    tk = t0
+    while (local_actor.is_in_line_of_sight(actor, pk.epoch(tk * pk.SEC2DAY))) and (tk - t0 < window_timeout_value_in_s):
+        tk += dt
+        transmitted_data_b += (
+            int(local_actor_comm_link.bandwidth_in_kbps * dt) * 1000
+        )  # (This is the quantum of information that you can transmit)
+
+    if tk - t0 >= window_timeout_value_in_s:
+        logger.debug(
+            "Timeout reached for the estimation of the communication window.")
+
+    return max(data_to_send_in_b - transmitted_data_b, 0), tk - t0
```

Comment:
Should we also tell the user total amount of data they can transmit? and also I think the need start and end of the comms window so that they can make an informed decision when to transfer how much?

Reply:
We can but it is data_to_send_in_b (input provided by the user) - max(data_to_send_in_b - transmitted_data_b, 0).
The start of the communication window is t0, for how it is implemented now (I don't think it makes sense returning it, because it is the user asking to start checking from that time). 
The end of the communication window is tk. So the it is t0 + communication windown (we can return it, but as for implemented now, it is again very straightforward to calculate t0 + communication_windows_s. )

Reply:
If I am a user what I want to know from paseos is at what time can I communicate with a satellite and how much data can I send. That allows me to plan my communications.
If you only give them duration from t0 and only how much can be transmitted the user has to repetitively query paseos to get this info.

Why not just return `(start_of_window,end_of_window,amount_of_data_that_can_be_transmitted)` ?
You have that info anyways, no need to keep it from the user. I am not sure we really need the info how much they want to transmit at this point? 🤔 This is more relevant in the activity when we must tell them if they request something impossible (like transmitting too much data)